nearcache/config.go

```diff
@@ -0,0 +1,256 @@
+	// Must be non-negative.
+	// The value 0 equals to math.MaxInt32
+	// The default is 0.
+	TimeToLiveSeconds int
+	//MaxIdleSeconds is the maximum number of seconds each entry can stay in the Near Cache as untouched (not-read).
+	//Entries that are not read (touched) more than MaxIdleSeconds value will get removed from the Near Cache.
+	//Accepts any integer between {@code 0} and {@link Integer#MAX_VALUE}.
+	// Must be non-negative.
+	// The value 0 equals to math.MaxInt32
+	// The default is 0.
+	MaxIdleSeconds int
+}
+
+// Clone returns a copy of the configuration.
+func (c Config) Clone() Config {
+	return Config{
+		invalidateOnChange: c.invalidateOnChange,
+		Name:               c.Name,
+		EvictionConfig:     c.EvictionConfig.Clone(),
+		PreloaderConfig:    c.PreloaderConfig.Clone(),
+		InMemoryFormat:     c.InMemoryFormat,
+		SerializeKeys:      c.SerializeKeys,
+		TimeToLiveSeconds:  c.TimeToLiveSeconds,
+		MaxIdleSeconds:     c.MaxIdleSeconds,
+	}
+}
+
+// Validate validates the configuration and replaces missing configuration with defaults.
+func (c *Config) Validate() error {
+	if c.Name == "" {
+		c.Name = "default"
+	}
+	if c.TimeToLiveSeconds == 0 {
+		c.TimeToLiveSeconds = math.MaxInt32
+	}
+	if c.MaxIdleSeconds == 0 {
+		c.MaxIdleSeconds = math.MaxInt32
+	}
+	if err := c.EvictionConfig.Validate(); err != nil {
+		return err
+	}
+	if err := c.PreloaderConfig.Validate(); err != nil {
+		return err
+	}
+	if _, err := check.NonNegativeInt32(c.TimeToLiveSeconds); err != nil {
+		return ihzerrors.NewInvalidConfigurationError("nearcache.Config.TimeToLiveSeconds: out of range", err)
+	}
+	if _, err := check.NonNegativeInt32(c.MaxIdleSeconds); err != nil {
+		return ihzerrors.NewInvalidConfigurationError("nearcache.Config.MaxIdleSeconds: out of range", err)
+	}
+	return nil
+}
+
+/*
+SetInvalidateOnChange sets if Near Cache entries are invalidated when the entries in the backing data structure are changed (updated or removed).
+When this setting is enabled, a Hazelcast instance with a Near Cache listens for cluster-wide changes on the entries of the backing data structure.
+And invalidates its corresponding Near Cache entries.
+Changes done on the local Hazelcast instance always invalidate the Near Cache immediately.
+Invalidate on change is true by default.
+*/
+func (c *Config) SetInvalidateOnChange(enabled bool) {
+	c.invalidateOnChange = &enabled
+}
+
+// InvalidateOnChange returns true invalide on change is enabled.
+// See the documentation for SetInvalidateOnChange.
+func (c Config) InvalidateOnChange() bool {
+	if c.invalidateOnChange == nil {
+		return true
+	}
+	return *c.invalidateOnChange
+}
+
+/*
+EvictionConfig is the configuration for eviction.
+
+You can set a limit for number of entries.
+The default values of the eviction configuration are:
+
+	* EvictionPolicyLRU as eviction policy
+	* MaxSizePolicyEntryCount as max size policy
+	* 2147483647 as maximum size for on-heap Map
```

Comment:
We can use the max int32 constant of go instead.

Reply:
I don't think that applies to `MaxSizePolicyEntryCount`, I'll remove that.

Reply:
Removed at: https://github.com/hazelcast/hazelcast-go-client/pull/786/commits/112f3ac833c9d97eb9261bb3b436431ac8177081